### PR TITLE
IBX-8356: Reworked `Ibexa\Core\MVC\Symfony\Security\Authentication\AuthenticatorInterface` usages to comply with Symfony-based authentication

### PR DESCRIPTION
**Pull request by konradoboza**
| :ticket: Issue | IBX-8356 |
|----------------|-----------|

#### Related PRs: 
- https://github.com/ibexa/recipes-dev/pull/124
- https://github.com/ibexa/core/pull/375

#### Description:
After going through multiple approaches we realized that `Authentication` mutation should **not** authenticate user. Instead we simply compare credentials without any security voodoo at this point and return newly generated JWT Token. From now on, it can be used as `Authorization: Bearer <token_here>` header to perform any operations in Ibexa Repository.

Example query:
**Generating token via GraphQL mutation (fully backwards compatible with what it was before):**
```
mutation CreateToken {
  createToken(username: "admin", password: "publish") {
    token
    message
  }
}
```
produces:
```
{
  "data": {
    "createToken": {
      "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE3MTkzMTk2MjIsImV4cCI6MTcxOTMyMzIyMiwicm9sZXMiOltdLCJ1c2VybmFtZSI6ImFkbWluIn0.PCPl-PhXU2RYEOSyr13usg4D_NyqijBt4CMO9smTpHU",
      "message": null
    }
  }
}
```
Then use the token as in:
```
curl -H "Content-Type: application/json" -H 'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE3MTkzMTk2MjIsImV4cCI6MTcxOTMyMzIyMiwicm9sZXMiOltdLCJ1c2VybmFtZSI6ImFkbWluIn0.PCPl-PhXU2RYEOSyr13usg4D_NyqijBt4CMO9smTpHU' --data @json.json http://localhost:8001/graphql
```
where `json.json` file might contain some query (to increase readability):
```
{
    "query": "mutation {createFolder(language: eng_GB, parentLocationId: 2, input: {name: \"ddd\"}) {id}}"
}
```

Response should look mor or less as follows:
```
{"data":{"createFolder":{"id":"RG9tYWluQ29udGVudDo="}}}
```
#### For QA:
All related to JWT in GraphQL needs to be checked.

#### Documentation:
We should probably mention dropped code even though query and their responses are the same.

### CODE REVIEW

**Review by alongosz — APPROVED** [review]

**Comment by adamwojs** [comment]
@konradoboza Could you please provide example of request with authentication using JWT?

**Comment by konradoboza** [comment]
@adamwojs since you cannot perform such queries via GraphiQL client, the only way is to make a cURL request with `Authorization: Bearer <here_goes_your_jwt_token>` formatted header. Unfortunately, I discovered some issues here, so making this one as work in progress again.

File: src/lib/Mutation/AuthenticationMutation.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * @return array<string, ?string>
+     *
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     */
+    public function createToken(Argument $args): array
+    {
+        $arguments = $args->getArrayCopy();
+        $username = $arguments['username'];
+        $password = $arguments['password'];
```
Comment (Steveb-p) [inline src/lib/Mutation/AuthenticationMutation.php:34]:
```suggestion
        $username = $arguments['username'] ?? null;
        $password = $arguments['password'] ?? null;
```
or check it in `isset` below instead.

Comment (konradoboza) [reply src/lib/Mutation/AuthenticationMutation.php:34]: Please see line 36.

Comment (Steveb-p) [reply src/lib/Mutation/AuthenticationMutation.php:34]: You're checking that **those variables** are set and not null.

In line 36 you are assigning values from an array, but that array does not necessarily contain those keys.

Comment (konradoboza) [reply src/lib/Mutation/AuthenticationMutation.php]: I misunderstood your comment, fixed via https://github.com/ibexa/graphql/compare/f43c65ec3a65bd6bce0f566bbf3faa34586ca37d..52e0d1ff5a80d75f62692edfb7a57069ee148f3c.

**Review by webhdx — APPROVED** [review]

**Review by ciastektk — APPROVED** [review]

File: src/lib/Mutation/AuthenticationMutation.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        } catch (NotFoundException) {
+            return [
+                'message' => 'Wrong username',
+                'token' => null,
+            ];
+        }
+
+        if (!$this->userService->checkUserCredentials($user, $args['password'])) {
+            return [
+                'message' => 'Wrong password',
```
Comment (Steveb-p) [inline src/lib/Mutation/AuthenticationMutation.php:50]: :warning: NO.

Never expose the difference that user exists and you're checking his password only.

**Review by Steveb-p — CHANGES_REQUESTED** [review]

**Review by Steveb-p — APPROVED** [review]

**Review by micszo — APPROVED** [review]

**Pull request merged**